web/app/components/input/input.tsx

```diff
@@ -148,10 +172,163 @@ export class Input extends Component<Props, State> {
       });
   }
 
-  render(props: RenderableProps<Props>, { isDisabled, isErrorShown, errorMessage, preview, maxLength, text }: State) {
+  /** appends error to input's error block */
+  appendError(...errors: string[]) {
```

Comment:
Could you move this login to HOC? Like this  http://uidevdaily.com/2018/react-drag-and-drop-file-upload-hoc/ ?

Reply:
Example you provided is overcomplicated for this current case. I don't need to bind to window, to dragleave, gragenter. Making it HOC assumes that I should provide props like `allowsMulptiple`, "allowsFiletype" and so on to make it generic for further reuse, which I think is meaningless now. If demand will appear, sure, it can be rewritten. But now let it be two methods.

Reply:
The main idea extract logic which responsible for upload file to separate function. It is not about reuse code or parametrize. Link was just example. I don't recommend it implement it as is.